components/dashboard/src/App.tsx

```diff
@@ -495,12 +498,26 @@ function App() {
         );
         return <div></div>;
     }
+    // Prefix with `/#referrer` will specify an IDE for workspace
+    // We don't need to show IDE preference in this case
+    const shouldUserIdePreferenceShown = User.isOnboardingUser(user) && !hash.startsWith(ContextURL.REFERRER_PREFIX);
+    if (shouldUserIdePreferenceShown !== showUserIdePreference) {
+        setShowUserIdePreference(shouldUserIdePreferenceShown);
```

Comment:
setting a state here will result in an additional render cycle, AFAIC.

you can simplify with a `const` here:
```
const showUserIdePreference = shouldUserIdePreferenceShown !== showUserIdePreference;
```

Reply:
@AlexTugarev Prev solution is because we have this one in App.tsx, maybe need to change it later?
https://github.com/gitpod-io/gitpod/blob/b29409611ef14ad4df628e24e309882391e90f8a/components/dashboard/src/App.tsx#L331-L334